experimental/unitsconversion/src/provider.rs

```diff
@@ -33,5 +33,76 @@ pub struct UnitsConstantsV1<'data> {
     // Also, the constant types.
     /// Maps from constant name (e.g. ft_to_m) to the value of the constant (e.g. 0.3048).
     #[cfg_attr(feature = "serde", serde(borrow))]
-    pub constants_map: ZeroMap<'data, str, str>,
+    pub constants_map: ZeroMap<'data, str, ConstantValueULE>,
+}
+
+/// This enum is used to represent the type of a constant value.
+/// It can be either `ConstantType::Actual` or `ConstantType::Approximate`.
+/// If the constant type is `ConstantType::Approximate`, it indicates that the value is not numerically accurate.
+#[zerovec::make_ule(ConstantExactnessULE)]
+#[cfg_attr(
+    feature = "datagen",
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_unitsconversion::provider),
+)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[derive(Copy, Clone, Debug, PartialOrd, Ord, PartialEq, Eq, Default)]
+#[repr(u8)]
+pub enum ConstantExactness {
+    #[default]
+    Actual = 0,
+    Approximate = 1,
+}
+
+/// This enum is used to represent the sign of a constant value.
+#[zerovec::make_ule(SignULE)]
+#[cfg_attr(
+    feature = "datagen",
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_unitsconversion::provider),
+)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[derive(Copy, Clone, Debug, PartialOrd, Ord, PartialEq, Eq, Default)]
+#[repr(u8)]
+pub enum Sign {
+    #[default]
+    Positive = 0,
+    Negative = 1,
+}
+
+/// This struct represents a constant value, which is composed of a numerator, denominator, sign, and type.
+#[zerovec::make_varule(ConstantValueULE)]
+#[derive(Clone, Debug, PartialOrd, Ord, PartialEq, Eq, Default)]
+#[cfg_attr(
+    feature = "datagen",
+    derive(databake::Bake),
+    databake(path = icu_unitsconversion::provider),
+)]
+#[cfg_attr(
+    feature = "datagen",
+    derive(serde::Serialize),
+    zerovec::derive(Serialize)
+)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Deserialize),
+    zerovec::derive(Deserialize)
+)]
+#[zerovec::derive(Debug)]
+pub struct ConstantValue<'data> {
+    // TODO(https://github.com/unicode-org/icu4x/issues/4092).
+    /// The numerator of the constant value in bytes starting with the least significant byte.
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub numerator: ZeroVec<'data, u8>,
+
+    // TODO(https://github.com/unicode-org/icu4x/issues/4092).
+    /// The denominator of the constant value in bytes starting with the least significant byte.
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub denominator: ZeroVec<'data, u8>,
+
+    /// Determines whether the constant value is positive or negative.
+    pub sign: Sign,
```

Comment:
thought: if you make one of the denominator/numerator signed, you only need a bit instead of an extra byte to store the sign.

Reply:
not only the sign, it is still the `ConstantExactness`. Therefore, I prefer to add just a todo

Reply:
https://github.com/unicode-org/icu4x/issues/4098

Reply:
The exactness will be harder to optimise away, whereas the sign has an obvious solution.

Reply:
What do you mean by consistency? This is just the storage format, `BigRational` internally uses two signed numbers.

Reply:
I lean toward storing two unsigned bigints and storing the sign separately. We can stick the sign bit in with the ConstantExactness in the ULE representation.

Also, how do you propose storing the signed bigint? BigInt::[to_bytes_le](https://docs.rs/num-bigint/latest/num_bigint/struct.BigInt.html#method.to_bytes_le) returns an unsigned byte vector and a sign enum.

Reply:
https://docs.rs/num-bigint/latest/num_bigint/struct.BigInt.html#method.to_signed_bytes_le

Reply:
How do you know when deserializing whether the number is positive or negative?

Like, how do you know if 0xFF is +255 or -1

Reply:
You deserialize with https://docs.rs/num-bigint/latest/num_bigint/struct.BigInt.html#method.from_signed_bytes_le?

Reply:
The implementation of that function seems nontrivial; it involves copying the bytes to a new vector and complementing them 

https://docs.rs/num-bigint/latest/src/num_bigint/bigint/convert.rs.html#381

Reply:
I think how the library serializes negatives is just by making the high bit in the highest nonzero byte be 1. But this means we lose capacity. For example, I imagine that you need to store 9 bytes for 255 (8 before

Reply:
Ah I assumed `BigInt` would use 2s complement internally, but it uses a sign bit as well. I guess this aligns with what Younies is currently doing.

Not that `from_bytes_le` *also allocates*, it's not possible to create a borrowed `BigUint`.